src/lib.rs

```diff
@@ -655,6 +674,22 @@ impl fmt::Debug for AstarteSdk {
     }
 }
 
+mod utils {
+    pub fn extract_purgeproperties(bdata: &[u8]) -> Vec<String> {
```

Comment:
I'd call this `extract_set_properties` since the returned properties here are those that are _not_ purged (and it's more readable with some kind of whitespace between words). `set` can also be `kept`, `valid`, `existing`, whatever sounds better to you to convey the concept.